Source/mpqapi.cpp

```diff
@@ -584,15 +584,15 @@ BOOL WriteMPQHeader()
 	DWORD NumberOfBytesWritten;
 
 	memset(&fhdr, 0, sizeof(fhdr));
-	fhdr.signature = '\x1AQPM';
-	fhdr.headersize = 32;
-	fhdr.filesize = GetFileSize(sghArchive, 0);
-	fhdr.version = 0;
-	fhdr.sectorsizeid = 3;
-	fhdr.hashoffset = 32872;
-	fhdr.blockoffset = 104;
-	fhdr.hashcount = 2048;
-	fhdr.blockcount = 2048;
+	fhdr.signature = SDL_SwapLE32('\x1AQPM');
+	fhdr.headersize = SDL_SwapLE32(32);
+	fhdr.filesize = SDL_SwapLE32(GetFileSize(sghArchive, 0));
+	fhdr.version = SDL_SwapLE16(0);
+	fhdr.sectorsizeid = SDL_SwapLE16(3);
+	fhdr.hashoffset = SDL_SwapLE32(32872);
+	fhdr.blockoffset = SDL_SwapLE32(104);
+	fhdr.hashcount = SDL_SwapLE32(2048);
+	fhdr.blockcount = SDL_SwapLE32(2048);
 
 	if (SetFilePointer(sghArchive, 0, NULL, FILE_BEGIN) == -1)

```

Comment:
Interestingly `fhdr` doesn't appear to be byte swapped until after getting the file pointer, this shouldn't however make a difference as `fhdr` is unused up to that point.